source/extensions/filters/http/peer_metadata/filter.cc

```diff
@@ -180,8 +180,11 @@ absl::optional<PeerInfo> MXMethod::lookup(absl::string_view id, absl::string_vie
 }
 
 MXPropagationMethod::MXPropagationMethod(
-    Server::Configuration::ServerFactoryContext& factory_context)
-    : id_(factory_context.localInfo().node().id()), value_(computeValue(factory_context)) {}
+    Server::Configuration::ServerFactoryContext& factory_context,
+    io::istio::http::peer_metadata::Config_IstioHeaders istioHeaders)
+    : id_(factory_context.localInfo().node().id()), value_(computeValue(factory_context)) {
+  skip_external_clusters_ = istioHeaders.skip_external_clusters();
```

Comment:
initialize in the constructor like `id_`, then you can make it `const`

Reply:
done